dataset/src/main/kotlin/org/jetbrains/kotlinx/dl/dataset/preprocessor/image/Cropping.kt

```diff
@@ -29,22 +30,27 @@ public class Cropping(
     public var left: Int = 1,
     public var right: Int = 1
 ) : ImagePreprocessor {
-    override fun apply(image: BufferedImage, inputShape: ImageShape): Pair<BufferedImage, ImageShape> {
-        val croppedImageShape =
-            ImageShape(
-                width = inputShape.width!! - left - right,
-                height = inputShape.height!! - top - bottom,
-                channels = inputShape.channels
-            )
+
+    override fun getOutputShape(inputShape: ImageShape?): ImageShape? {
```

Comment:
Looks like all transformation should have implementation and ability to return/control shape.